packages/legend-extension-application-studio-query-builder/src/components/ServiceQueryBuilder.tsx

```diff
@@ -92,6 +97,11 @@ export const ServiceQueryBuilder = observer(
                             queryBuilderExtension.queryBuilderState.changeDetectionState.setQueryHashCode(
                               hashObject(rawLambda),
                             );
+                            await flowResult([
```

Comment:
not sure why we need to do this since we're closing the service query editor anyway

Reply:
Maybe let's remove all change detection logic here altogether